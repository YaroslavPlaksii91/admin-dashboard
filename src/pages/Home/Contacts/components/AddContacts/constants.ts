import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const ADD_CONTACT_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const ADD_CONTACT_CONFIG = Object.freeze({
  [ADD_CONTACT_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [ADD_CONTACT_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [ADD_CONTACT_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [ADD_CONTACT_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [ADD_CONTACT_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
