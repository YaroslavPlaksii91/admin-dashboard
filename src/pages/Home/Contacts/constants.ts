export const CONTACTS_COLUMNS = Object.freeze([
  {
    title: 'Name',
    name: 'name',
    size: 4,
  },
  {
    title: 'Email',
    name: 'email',
    size: 3,
  },
  {
    title: 'Address',
    name: 'address',
    size: 3,
  },
  {
    title: 'Created at',
    name: 'created',
    size: 2,
  },
]);
