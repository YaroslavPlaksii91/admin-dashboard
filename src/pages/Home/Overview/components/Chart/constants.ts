export const TOOLTIP_TEST_ID = 'tooltip';
