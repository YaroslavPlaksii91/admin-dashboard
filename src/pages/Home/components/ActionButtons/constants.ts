export const SORT_BTN_TEST_ID = 'sortButton';
export const FILTER_BTN_TEST_ID = 'filterButton';
export const ADD_BTN_TEST_ID = 'addButton';
