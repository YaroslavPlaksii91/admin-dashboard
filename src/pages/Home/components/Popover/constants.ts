export const POPOVER_TEST_ID = 'popover';
