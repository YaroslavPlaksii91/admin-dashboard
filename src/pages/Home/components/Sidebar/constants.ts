export const USERS_PAGES = Object.freeze([
  {
    name: 'Overview',
    icon: '/src/assets/icons/sprite.svg#icon-overview',
    link: '/overview',
  },
  {
    name: 'Tickets',
    icon: '/src/assets/icons/sprite.svg#icon-tickets',
    link: '/tickets',
  },
  {
    name: 'Ideas',
    icon: '/src/assets/icons/sprite.svg#icon-ideas',
    link: '/ideas',
  },
  {
    name: 'Contacts',
    icon: '/src/assets/icons/sprite.svg#icon-contacts',
    link: '/contacts',
  },
  {
    name: 'Agents',
    icon: '/src/assets/icons/sprite.svg#icon-agents',
    link: '/agents',
  },
  {
    name: 'Articles',
    icon: '/src/assets/icons/sprite.svg#icon-articles',
    link: '/articles',
  },
]);

export const USERS_SETTINGS = Object.freeze([
  {
    name: 'Settings',
    icon: '/src/assets/icons/sprite.svg#icon-settings',
    link: '/settings',
  },
  {
    name: 'Subscription',
    icon: '/src/assets/icons/sprite.svg#icon-subscription',
    link: '/subscription',
  },
]);

export const SIDEBAR_LINK_TEST_ID = 'sidebarLink';
