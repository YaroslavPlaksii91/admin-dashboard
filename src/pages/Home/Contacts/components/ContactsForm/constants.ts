import { IMAGE_SIZE } from '@components/AddPhoto/constants';

export const CONTACTS_FORM_FIELDS = Object.freeze({
  PHOTO: 'photo',
  FIRST_NAME: 'firstName',
  LAST_NAME: 'lastName',
  EMAIL: 'email',
  ADDRESS: 'address',
});

export const CONTACTS_FORM_CONFIG = Object.freeze({
  [CONTACTS_FORM_FIELDS.PHOTO]: {
    label: 'Add photo',
    name: 'photo',
    size: IMAGE_SIZE.BIG,
  },
  [CONTACTS_FORM_FIELDS.FIRST_NAME]: {
    label: 'first name',
    type: 'text',
    name: 'firstName',
    placeholder: 'First name',
  },
  [CONTACTS_FORM_FIELDS.LAST_NAME]: {
    label: 'last name',
    type: 'text',
    name: 'lastName',
    placeholder: 'Last name',
  },
  [CONTACTS_FORM_FIELDS.EMAIL]: {
    label: 'email',
    type: 'email',
    name: 'email',
    placeholder: 'Email',
  },
  [CONTACTS_FORM_FIELDS.ADDRESS]: {
    label: 'address',
    type: 'text',
    name: 'address',
    placeholder: 'Address',
  },
});
