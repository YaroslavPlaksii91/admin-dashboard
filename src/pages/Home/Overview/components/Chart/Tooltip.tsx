import { FC } from 'react';
import { SvgIcon } from '@mui/material';

import { TooltipProps } from './types';
import { TOOLTIP_TEST_ID } from './constants';

export const CustomTooltip: FC<TooltipProps> = ({ active, payload }) => {
  if (active && payload?.length) {
    const value = payload[1].value;

    return (
      <SvgIcon
        sx={{
          width: 64,
          height: 64,
          filter: 'drop-shadow(0px 1px 4px rgba(0, 0, 0, 0.08))',
        }}
        data-testid={TOOLTIP_TEST_ID}
      >
        <use href="/sprite.svg#icon-sheet"></use>
        <text x="9" y="12" fill="#252733" fontSize={5} fontWeight={600}>
          {value}
        </text>
      </SvgIcon>
    );
  }

  return null;
};
