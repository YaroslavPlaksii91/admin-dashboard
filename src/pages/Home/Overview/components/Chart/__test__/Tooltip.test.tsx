import { render, screen } from '@testing-library/react';

import { CustomTooltip } from '../Tooltip';
import { TOOLTIP_TEST_ID } from '../constants';

describe('CustomTooltip component', () => {
  const payload = [{ value: '5' }, { value: '10' }];

  it('makes snapshot', () => {
    const { asFragment } = render(
      <CustomTooltip active={true} payload={payload} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when active and payload is provided', () => {
    render(<CustomTooltip active={true} payload={payload} />);

    const svgIcon = screen.getByTestId(TOOLTIP_TEST_ID);
    expect(svgIcon).toBeInTheDocument();

    const textElement = screen.getByText('10');
    expect(textElement).toBeInTheDocument();
  });

  it('should not render when active is false', () => {
    render(<CustomTooltip active={false} payload={[]} />);

    const svgIcon = screen.queryByTestId(TOOLTIP_TEST_ID);
    expect(svgIcon).not.toBeInTheDocument();
  });

  it('should not render when payload is empty', () => {
    render(<CustomTooltip active={true} payload={[]} />);

    const svgIcon = screen.queryByTestId(TOOLTIP_TEST_ID);
    expect(svgIcon).not.toBeInTheDocument();
  });
});
