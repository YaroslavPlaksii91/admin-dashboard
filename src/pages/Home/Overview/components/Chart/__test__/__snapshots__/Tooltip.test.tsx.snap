// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip component makes snapshot 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-sp8ucn-MuiSvgIcon-root"
    data-testid="tooltip"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <use
      href="/src/assets/icons/sprite.svg#icon-sheet"
    />
    <text
      fill="#252733"
      font-size="5"
      font-weight="600"
      x="9"
      y="12"
    >
      10
    </text>
  </svg>
</DocumentFragment>
`;
